packages/cli/test/models/network/NetworkController.test.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+require('../../setup');
+
+import sinon from 'sinon';
+import NetworkController from '../../../src/models/network/NetworkController';
+import NetworkFile from '../../../src/models/files/NetworkFile';
+import ProjectFile from '../../../src/models/files/ProjectFile';
+import { AppProjectDeployer, ProxyAdminProjectDeployer } from '../../../src/models/network/ProjectDeployer';
+
+contract('NetworkController', function() {
+
+  let deployer, projectFile, networkFile, controller
+
+  beforeEach(()=> {
+    deployer = sinon.createStubInstance(AppProjectDeployer)
+  })
+
+  afterEach(() => {
+    sinon.restore();
+  })
+
+  describe('_getAllSolidityLibNames()', () => {
+    let controllerMock
+    beforeEach(() => {
+      projectFile = new ProjectFile('test/mocks/mock-stdlib/zos.json');
+      networkFile = new NetworkFile(projectFile, 'test');
+      controller = new NetworkController('test', {}, networkFile);
+      controllerMock = sinon.mock(controller)
+    })
+
+    it('should order dependencies correctly', () => {
+
+      const de = sinon.match.array.deepEquals
+
+      const contractNames = ['A', 'B', 'C']
+
+      controllerMock.expects('_getSolidityLibNames').withArgs(de(contractNames)).returns(['E', 'F', 'D'])
+      controllerMock.expects('_getSolidityLibNames').withArgs(de(['E', 'F', 'D'])).returns(['D', 'G', 'H'])
+      controllerMock.expects('_getSolidityLibNames').withArgs(de(['G', 'H'])).returns(['D', 'I'])
+      controllerMock.expects('_getSolidityLibNames').withArgs(de(['I'])).returns([])
+
+      const result = controller._getAllSolidityLibNames(['A', 'B', 'C'])
+
+      assert.deepEqual(result, ['I', 'D', 'G', 'H', 'E', 'F'])
```

Comment:
And the test has to be a bit different because of it does a full walk.
```
controllerMock
        .expects('_getSolidityLibNames')
        .withArgs(de(contractNames))
        .returns(['E', 'F', 'D']);
      controllerMock
        .expects('_getSolidityLibNames')
        .withArgs(de(['E', 'F', 'D']))
        .returns(['D', 'G', 'H']);
      controllerMock
        .expects('_getSolidityLibNames')
        .withArgs(de(['D', 'G', 'H']))
        .returns(['D', 'I']);
      controllerMock
        .expects('_getSolidityLibNames')
        .withArgs(de(['D', 'I']))
        .returns([]);

      const result = controller._getAllSolidityLibNames(['A', 'B', 'C']);

      assert.deepEqual(result, ['D', 'I', 'G', 'H', 'E', 'F']);
```